entities/IoCContainer.java

```diff
@@ -0,0 +1,34 @@
+package entities;
+
+import java.util.HashMap;
+import java.util.Map;
+/*
+Lines for use the container
+1. IoCContainer ioc = IoCContainer.getInstance();
+2. ioc.Register(Entitie_name,new Entitie_name());
+3. Entitie_name any_name = (Entitie_name)ioc.Resolve("Entitie_name");
+4. Now you can use the name used in any_name for to the methodes of the entitie
+ */
+public class IoCContainer<T> {
+
+    //Singleton to instantiate the IoCContainer only once
+    private static IoCContainer ioc;
+    private IoCContainer(){}
+    public static IoCContainer getInstance(){
+        if(ioc == null){
+            ioc = new IoCContainer();
+        }return ioc;
+    }
+
+    //IoCContainer structure
+    private final Map<String, T> map = new HashMap<String,T>();
+
+    public void Register(String s, T t){
+        map.put(s,t);
+    }
+
+    public T Resolve(String s){
+        return (T)map.get(s);
```

Comment:
Shouldn't you validate if the key already exists and if not then throw an error?

Reply:
What if you pass down a null key instead? shouldn't we validate that as well?

Reply:
Done